# SpinalCordInjury https://zenodo.org/badge/513728372.svg

<div align=left><img src="Figure 1 V11.jpg" width="1000"></div>  
  
### Highlight
  
  
- Reveal spatiotemporal transcriptional changes at gene expression and cellular response level after SCI

- Discover and identify a spatial-dependent neural protective state of activated Astrocyte-WM after injury

- Generate a database of gene expression after SCI at multi-dimension (time, distance, direction, anatomy, cell type)
  
 
### *Abstract*
  
<p>Although complex alterations in gene expression and molecular signaling in mammalian spinal cord injury (SCI) are documented, little is known about the anatomy of these gene activities and their changes. Here, we conduct a spatiotemporal transcriptomic profiling of gene expression in a mouse SCI model, which demonstrates comprehensive gene co-expression networks and regulatory programs in the morphological context of the spinal cord. Further, by integrating publicly available single-cell (sc) RNA sequencing (RNA-seq) data, our analysis reveals a delicate and exquisite orchestration of multicellular transcriptional programs and cell-to-cell signaling following SCI, leading to interesting new discoveries that were unappreciated in previous bulk or scRNA-seq studies. For example, the spatial view of the gene expressional changes identifies a distinct, SCI-induced group of astrocytes and the transcriptional programs underpinning the molecular and cellular alterations in the functional state of these cells are defined. Thus, the emerging spatial transcriptomics approach proves mighty not only to characterize developmental processes and chronic diseases but also to investigate acute neural injury. Together, this work provides a molecular atlas for understanding the spatial organization and transcriptional landscape underlying the perplexing, multicellular responses in mammalian SCI.<p>

### Cite: <br>
*Spatiotemporal orchestration of multicellular transcriptional programs and molecular signalings in spinal cord injury*
  
  
  
  
### Correspondence:<br>
Yanshan Fang ([fangys@sioc.ac.cn](mailto:fangys@sioc.ac.cn)); Guangdun Peng ([peng_guangdun@gibh.ac.cn](mailto:peng_guangdun@grmh-gdl.cn)); Ang Li([anglijnu@jnu.edu.cn](mailto:anglijnu@jnu.edu.cn))
  
  
  
